feat: Centered floating toolbar with auto-hide functionality

PrayagCodes opened the pull request:
## Summary

This pull request transforms the top toolbar into a centered floating island design with intelligent auto-hide functionality. The implementation provides users with increased screen real estate and a modern interface while maintaining full toolbar functionality.

## Closes Feature

Closes [Issue #4 - Implement dynamic top toolbar#8 ](https://github.com/codepath/puter/issues/8)
Closes [Issue #13 - Implement toolbar auto-hide #56 ](https://github.com/codepath/puter/issues/56)

## Changes


### Visual Design Updates
- Toolbar is now horizontally centered at the top of the screen
- Implements island design with auto-width and 20px border-radius
- Positioned 10px from top edge with floating shadow effect
- Button spacing updated to 5px gap with 15px horizontal padding
- Logo margin removed to support centered layout

### Auto-Hide Implementation
- Automatically hides after 2 seconds of mouse inactivity
- Reveals when mouse cursor moves within 50px of screen top
- Implements peek mode with 60% opacity on mouse proximity
- Adjusts desktop and window layout when toolbar visibility changes
- Includes settings toggle in Personalization tab
- Synchronizes preferences between localStorage and KV store with fallback

### Technical Architecture
- Class-based `ToolbarAutoHide` singleton for state management
- CSS animation states: `toolbar-visible`, `toolbar-hidden`, `toolbar-peek`
- Combined CSS transforms maintain horizontal centering during animations
- Implements error handling with graceful fallback mechanisms
- Includes double-initialization protection
- Self-contained preference loading without external dependencies

### Detailed Changes by File

**src/gui/src/css/style.css**
- Added centered island toolbar styles
- Implemented auto-hide animation states
- Added toggle switch component styles

**src/gui/src/helpers/toolbar_autohide.js**
- New file containing ToolbarAutoHide class
- State management and animation logic
- Preference loading and synchronization
- Mouse proximity detection

**src/gui/src/UI/UIDesktop.js**
- Removed logo margin-right auto property
- Integrated auto-hide initialization

**src/gui/src/UI/Settings/UITabPersonalization.js**
- Added auto-hide toggle switch control
- Implemented preference load/save handlers

**src/gui/src/globals.js**
- Added toolbar_autohide default preference (false)

PrayagCodes committed:
feat(ui): add centered floating toolbar with auto-hide

  Transform toolbar into a modern centered island design with smart
  auto-hide functionality for maximizing workspace.

  Features:
  - Centered floating toolbar (macOS-style)
  - Auto-hide after 2 seconds of inactivity
  - Reveals on mouse proximity to top edge
  - Settings toggle in Personalization
  - Smooth animations and peek mode
  - Automatic layout adjustments

  Technical:
  - Class-based ToolbarAutoHide singleton
  - CSS transform states for animations
  - Preference sync (localStorage + KV store)
  - Self-contained with error handling

israx commented on src/gui/src/UI/Settings/UITabPersonalization.js:
```diff
@@ -102,6 +110,42 @@ export default {
+        
+        // Load the current state from KV store and set checkbox
+        puter.kv.get('user_preferences.toolbar_autohide').then(async (val) => {
+            const isEnabled = val === true || val === 'true';
```
when would be the case when `val` is a string?

PrayagCodes committed:
refactor: use idiomatic boolean conversion for toolbar_autohide preference

Changed from al === true || val === 'true' to !!val in both
UITabPersonalization.js and toolbar_autohide.js.

The KV store uses JSON serialization which preserves boolean types,
making the string 'true' check unnecessary. The !!val conversion
correctly handles true, false, undefined, and null values while
being more idiomatic JavaScript.

PrayagCodes committed:
fix: prevent toolbar from hiding when context menu is open

When toolbar auto-hide is enabled and a user opens a context menu
(e.g., user profile menu), the toolbar was hiding after 2 seconds
while the menu remained visible, leaving the menu orphaned.

This fix:
- Adds hasOpenMenu() method to detect open context menus from toolbar
- Prevents toolbar from hiding when any toolbar-related menu is open
- Listens for ctxmenu-will-open event to immediately prevent hiding
- Adds periodic check to keep toolbar visible while menus are open
- Updates all hide timer logic to check for open menus before hiding

Fixes issue where clicking user profile icon would open menu, but
toolbar would hide leaving the menu visible but disconnected.
